pac4j-core/src/main/java/org/pac4j/core/context/ExcludePathMatcher.java

```diff
@@ -0,0 +1,71 @@
+/*
+ *    Copyright 2012 - 2015 pac4j organization
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package org.pac4j.core.context;
+
+import org.pac4j.core.exception.TechnicalException;
+import org.pac4j.core.util.CommonHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.regex.Pattern;
+
+/**
+ * Exclude path matcher.
+ *
+ * @author Jerome Leleu
+ * @since 1.8.1
+ */
+public final class ExcludePathMatcher implements PathMatcher {
+
+    private final static Logger logger = LoggerFactory.getLogger(ExcludePathMatcher.class);
+
+    private String excludePath;
+
+    private Pattern pattern;
+
+    public ExcludePathMatcher() {
+    }
+
+    public ExcludePathMatcher(final String excludePath) {
+        setExcludePath(excludePath);
+    }
+
+    public boolean matches(final WebContext context) {
+        if (pattern != null) {
+            final String path = context.getPath();
+            logger.debug("path to match: {}", path);
+            return pattern.matcher(path).matches();
+        }
+        return false;
+    }
+
+    public String getExcludePath() {
+        return excludePath;
+    }
+
+    public void setExcludePath(String excludePath) {
+        this.excludePath = excludePath;
+        if (CommonHelper.isNotBlank(excludePath)) {
+            logger.warn("Excluding paths is an advanced feature: be careful when defining your regular expression to avoid any security issue!");
+            if (!excludePath.startsWith("^") || !excludePath.endsWith("$")) {
```

Comment:
That's a good question. At first, I wanted to allow people to define the regexp they want to match a path, but I realized it could be very dangerous.

Let's say I have two pages: `unsecure.html` and `secure.html`. I put a filter on both and want to exclude one with `unsecure\.html` as the `excludePath`. It works perfectly and it feels good, but it's a **security breach**.

Because, as an attacker, I can call `secure.html?attack=unsecure.html` and it matches the exclusion path and no security applies!

So I decide to force a full path starting with `^`  and ending with `$`.  If you use `^unsecure\.html$` as the `excludePath`, it works without any security issue.